src/isoStringScalar/isoStringScalar.ts

```diff
@@ -0,0 +1,86 @@
+import { GraphQLScalarType, Kind } from 'graphql';
+import { isValid, parseISO, toDate } from 'date-fns';
+import {
+  InternalServerError,
+  UserInputError,
+} from '../errorHandler/errorHandler';
+
+export const isoStringScalar = new GraphQLScalarType({
+  name: 'ISOString',
+  description: 'ISOString custom scalar type',
+
+  /**
+   * Converts server-side TS Date representation to a JSON-compatible, ISO-8601-compliant
+   * UTC Datetime String for Apollo Server to include in an operation response.
+   *
+   * @param value - TS Date Object as generated by Data Store Client & validated by mysqlDateConvert.
+   * @returns ISO-8601-compliant, UTC-based Datetime String.
+   */
+  serialize(value: Date | null): string | null {
+    if (value === null) {
+      return null;
+    }
+
+    if (value instanceof Date && isNaN(value.valueOf())) {
+      throw new InternalServerError(
+        'Invalid Data Store Response: invalid Date object'
+      );
+    }
+
+    if (value instanceof Date) {
```

Comment:
Optional style: invert this and place it immediately after the null check, throwing an `InternalServerError` if `value` isn't a `Date`.

I like to think that I am building a case that `value` is a valid date value, and you can rely on facts that you've already previously validated.  After that step, you've already validated that value is a date, so `value instanceof Date &&` can be removed from the NaN check, and your happy path code isn't in a branch.

Reply:
that makes sense, i like that approach. i think i did what you proposed, sorry if i misunderstood though